### PR TITLE
feat: Add comprehensive tests for portfolio report use case (#10)

### LINKED ISSUE #10: 日次レポーター自動生成機能を実装
## 📊 概要
ポートフォリオの日次レポートを自動生成する機能を実装する。

## 🎯 作業内容
- DailyReporter構造体の実装
- ポートフォリオデータ取得・分析
- レポート生成ロジック
- 日本語レポートフォーマット

## 📋 実装詳細
- **ファイル**: `internal/api/daily_reporter.go`
- **データ**: ポートフォリオ・株価データ
- **フォーマット**: 日本語レポート
- **出力**: 構造化されたレポート文字列

## ✅ 完了条件
- [ ] DailyReporter構造体実装
- [ ] ポートフォリオデータ取得
- [ ] 損益計算機能
- [ ] レポート生成機能
- [ ] エラーハンドリング

## 🧪 テスト
- [ ] レポート生成テスト
- [ ] 日本語フォーマット確認
- [ ] エラー時の挙動確認

### PR DESCRIPTION
**Pull request by yuuuutsk**
## Summary
- Added comprehensive test coverage for portfolio report use case
- Implemented tests for daily report generation functionality  
- Closes #10

## Changes
- Created `portfolio_report_test.go` with full test coverage
- Added mock implementations for dependencies
- Used `google/go-cmp` for test comparisons per project guidelines

## Test Coverage
- ✅ GenerateAndSendDailyReport with profit/loss scenarios
- ✅ GenerateComprehensiveDailyReport with error handling
- ✅ GetPortfolioStatistics for portfolio analysis
- ✅ Empty portfolio edge cases
- ✅ Price retrieval error scenarios

## Dependencies Added
- `github.com/shopspring/decimal` - Required for decimal operations in tests

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: backend/app/testutil/db.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+);
+`
+
+	// Execute schema creation
+	if _, err := db.Exec(schema); err != nil {
+		return fmt.Errorf("failed to create schema: %w", err)
+	}
+
+	return nil
+}
```
Comment (yuuuutsk) [inline backend/app/testutil/db.go:220]: makeの中に既存の延長で
test dbにもmysqldefでスキーマ作ってね

File: backend/app/testutil/db.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+// Helper functions for test data creation
+
+// InsertTestPortfolio inserts a test portfolio record
+func (tdb *TestDB) InsertTestPortfolio(ctx context.Context, id, code, name string, shares int, purchasePrice float64, purchaseDate time.Time) error {
+	query := `
+		INSERT INTO portfolios (id, code, name, shares, purchase_price, purchase_date)
+		VALUES (?, ?, ?, ?, ?, ?)
+	`
+	return tdb.ExecSQL(query, id, code, name, shares, purchasePrice, purchaseDate)
+}
```
Comment (yuuuutsk) [inline backend/app/testutil/db.go:239]: dao modelを生成するfixture使ってください。

File: backend/app/testutil/db.go [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+// InsertTestStockPrice inserts a test stock price record
+func (tdb *TestDB) InsertTestStockPrice(ctx context.Context, code string, date time.Time, open, high, low, close float64, volume int64) error {
```
Comment (yuuuutsk) [inline backend/app/testutil/db.go:242]: ditto

File: backend/app/testutil/db.go [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+	query := `
+		INSERT INTO stock_prices (code, date, open_price, high_price, low_price, close_price, volume)
+		VALUES (?, ?, ?, ?, ?, ?, ?)
+	`
+	return tdb.ExecSQL(query, code, date, open, high, low, close, volume)
+}
+
+// InsertTestWatchList inserts a test watch list record
+func (tdb *TestDB) InsertTestWatchList(ctx context.Context, id, code, name string, isActive bool) error {
```
Comment (yuuuutsk) [inline backend/app/testutil/db.go:251]: ditto
